dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/query/QueryServiceTest.java

```diff
@@ -621,6 +621,15 @@ void testCriteriaOrRootJunctionDE() {
     assertEquals(3, objects.size());
   }
 
+  /**
+   * Hibernate should create a join to dataelement table and all the equal operators should use that
+   * join. Expected query generated: {@code select dataelement.* from dataelementgroup dataelemen0_
+   * inner join dataelementgroupmembers members1_ on
+   * dataelemen0_.dataelementgroupid=members1_.dataelementgroupid inner join dataelement
+   * dataelemen2_ on members1_.dataelementid=dataelemen2_.dataelementid where dataelemen2_.uid=? and
+   * dataelemen2_.uid=? and dataelemen2_.uid=? and dataelemen2_.uid=? and dataelemen2_.uid=? and
+   * dataelemen2_.uid=? limit ?}
+   */
```

Comment:
This is good information, but having an expected result in a comment and not in the test doesn't seem right. What if the actual query doesn't mirror the expectation? (we'll never know)

Reply:
yes, this was a reminder for myself. Replaced with assertion now.